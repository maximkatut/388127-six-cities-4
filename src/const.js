export const OfferTypes = [
  `Apartment`, `Room`, `House`, `Hotel`
];

// MAP CONSTS
export const MAP_ZOOM = 12;
export const MAP_ICON_SIZE = [27, 39];
export const MAP_ICON_URL = `img/pin.svg`;
