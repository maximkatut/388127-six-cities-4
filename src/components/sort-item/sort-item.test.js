import React from 'react';
import renderer from 'react-test-renderer';
import SortItem from './sort-item.jsx';
import {SortType} from '../../const.js';

describe(`SortItem`, () => {

  let component;
  const onSortClick = jest.fn();

  beforeEach(() => {

    component = renderer.create(
        <SortItem
          sortType={SortType.POPULAR}
          onSortClick={onSortClick}
        />
    );
  });

  it(`SortItem component should render correctly`, () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it(`SortItem should dispatch an action on sort-menu dropdown click`, () => {
    renderer.act(() => {
      component.root.findAllByType(`li`)[0].props.onClick();
    });

    expect(onSortClick).toHaveBeenCalledTimes(1);
  });
});
