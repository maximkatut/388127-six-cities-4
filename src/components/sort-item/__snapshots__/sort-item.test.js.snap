// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortItem SortItem component should render correctly 1`] = `
<li
  className="places__option "
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  tabIndex={0}
>
  Popular
</li>
`;
